src/TableV8/index.tsx

```diff
@@ -17,84 +17,159 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table';
 import {
-  StyledTableContainer,
-  StyledTableHeader,
-  StyledTable,
-  StyledTableBody,
-  StyledTableCell,
-  StyledTableHead,
-  StyledTableRow,
-  StyledResizableTableHeaderContent,
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+import { useState } from 'react';
+import {
+  Loader,
+  Resizer,
+  SortButton,
+  TableStyled,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableCellWrapper,
+  TableHeaderWrapper,
 } from './styled';
 
+// IMPORTANT
+// react table v8 is headless and we made our own UI.
+// see the readme & stories for help.
+
+declare module '@tanstack/table-core' {
+  interface ColumnMeta<TData extends unknown, TValue> {
+    customCell?: boolean;
+    customHeader?: boolean;
+  }
+}
+
 interface ReactTableProps<TData> {
   className?: string;
   columns: ColumnDef<TData>[];
   data: TData[];
+  LoaderComponent?: any;
+  loading?: boolean | null;
+  manualSorting?: boolean;
   withHeaders?: boolean;
+  withResize?: boolean;
   withRowBorder?: boolean;
   withRowHighlight?: boolean;
   withSideBorders?: boolean;
+  withSorting?: boolean;
   withStripes?: boolean;
 }
 
 export const TableV8 = <TData extends object>({
   className = '',
   columns = [],
   data = [],
+  LoaderComponent = Loader,
+  loading = false,
+  manualSorting = false,
   withHeaders = false,
-  withSideBorders = false,
+  withResize = false,
+  withRowBorder = false,
   withRowHighlight = false,
+  withSideBorders = false,
+  withSorting = false,
   withStripes = false,
-  withRowBorder = false,
 }: ReactTableProps<TData>) => {
+  const [sortingState, setSortingState] = useState<SortingState>([]);
+
   const table = useReactTable({
+    columnResizeMode: 'onChange',
     columns,
     data,
+    enableColumnResizing: withResize,
+    enableSorting: withSorting,
     getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    manualSorting,
+    onSortingChange: setSortingState,
+    state: {
+      sorting: sortingState,
+    },
```

Comment:
This sorting mechanism works on the front end, right?
How does this integration work when you do server side sorting? (as it is likely our most common use)

Reply:
the tables I've done so far have front end sorting. for the remaining tables I'm going to re-use logic from the donor summary table on the program dashboard that handles pagination, sorting, and filtering, and pass down state/handlers to TableV8. I might be able to turn that into a reducer as an enhancement. not sure if that'll work with React Table's API.